In [ ]:
%pip install iterative-stratification==0.1.9

# 데이터 로드

In [ ]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np
from iterstrat.ml_stratifiers import MultilabelStratifiedKFold

#데이터패스 수정해주세요
#DATA_PATH = "/content/drive/MyDrive/dataton/"
DATA_PATH = '/content/drive/MyDrive/dayoon/데이터톤/공유 데이터셋 및 베이스라인코드/'
SEED = 42
df = pd.read_csv(f'{DATA_PATH}dataset_0110.csv')
df.shape

<ipython-input-3-86db79c7b5c3>:9: DtypeWarning: Columns (37,103,109,207) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(f'{DATA_PATH}dataset_0110.csv')


(55989, 258)

In [ ]:
columns_to_select = [
    'id', 'id_fam', 'year', 'region', 'town_t', 'apt_t', 'sex', 'age', 'incm', 'ho_incm',
    'incm5', 'ho_incm5', 'edu', 'occp', 'cfam', 'genertn', 'allownc', 'house', 'live_t',
    'ainc_1', 'ainc', 'marri_1', 'marri_2', 'tins', 'npins', 'id_f', 'id_m', 'd_1_1', 'd_2_1',
    'd_2_wk', 'bh9_11', 'bh2_61', 'ac1_yr', 'ac3', 'ac3_1_01', 'ac3_2_01', 'ac8_1_01',
    'ac8_1e_01', 'ac3_1_02', 'ac3_2_02', 'ac8_1_02', 'ac8_1e_02', 'ac3_1_03', 'ac3_2_03',
    'ac8_1_03', 'ac8_1e_03', 'educ', 'ec_pedu_1', 'ec_pedu_2', 'ec1_1', 'ec_occp', 'ec_stt_1',
    'ec_stt_2', 'ec_wht_0', 'ec_wht_23', 'ec_lgw_2', 'ec_lgw_4', 'ec_lgw_5', 'bo1', 'bo1_1',
    'bo1_2', 'bo1_3', 'bo2_1', 'bd1', 'bd2', 'bd1_11', 'bd2_1', 'bd2_31', 'bd7_4', 'bd7_5',
    'ba2_12', 'ba2_13', 'ba2_2_1', 'ba2_2_2', 'ba2_2_3', 'ba2_2_4', 'ba2_2_5', 'ba2_2_6',
    'ba2_22', 'bp1', 'bp5', 'bp6_10', 'bp6_2', 'bp6_31', 'bp7', 'bs1_1', 'bs2_1', 'bs3_1',
    'bs3_2', 'bs3_3', 'bs6_2', 'bs6_2_1', 'bs6_2_2', 'bs6_3', 'bs6_4', 'bs6_4_1', 'bs6_4_2',
    'bs12_1', 'bs12_2', 'bs12_31', 'bs12_32', 'bs12_33', 'bs12_34', 'bs12_35', 'bs12_36',
    'bs12_41', 'bs12_42', 'bs12_43', 'bs12_44', 'bs12_45', 'bs12_46', 'bs5', 'bs5_1', 'bs8_2',
    'bs9_2', 'bs13', 'bs10_1', 'bs2', 'bs10_2', 'bs10_3', 'be3_71', 'be3_72', 'be3_73', 'be3_74',
    'be3_81', 'be3_82', 'be3_83', 'be3_84', 'be3_91', 'be3_92', 'be3_93', 'be3_94', 'be3_75',
    'be3_76', 'be3_77', 'be3_78', 'be3_85', 'be3_86', 'be3_87', 'be3_88', 'be8_1', 'be8_2',
    'be3_31', 'be3_32', 'be3_33', 'be5_1', 'lw_ms', 'lw_mp_e', 'lw_mp_a', 'lw_ms_a', 'lw_pr',
    'lw_pr_1', 'lw_mt', 'lw_mt_a1', 'lw_mt_a2', 'lw_br', 'lw_br_ch', 'lw_br_dur', 'lw_oc',
    'he_mens', 'he_prg', 'he_dprg', 'he_fh', 'he_hpfh1', 'he_hpfh2', 'he_hpfh3', 'he_hlfh1',
    'he_hlfh2', 'he_hlfh3', 'he_ihdfh1', 'he_ihdfh2', 'he_ihdfh3', 'he_strfh1', 'he_strfh2',
    'he_strfh3', 'he_dmfh1', 'he_dmfh2', 'he_dmfh3', 'he_rpls', 'he_ht', 'he_wt', 'he_wc',
    'he_bmi', 'he_obe', 'bm1_0', 'bm1_1', 'bm1_2', 'bm1_3', 'bm1_4', 'bm1_5', 'bm1_6', 'bm1_7',
    'bm1_8', 'bm7', 'bm8', 'or1_2', 'mo4_00', 'mo4_4', 'mo4_9', 'mo4_7', 'mo4_8', 'mo4_11',
    'mo4_12', 'mo4_18', 'mo4_15', 'bm14', 'bm14_1', 'bm14_2', 't_q_hr', 't_nq_ocp', 't_q_vn',
    'l_br_fq', 'l_ln_fq', 'l_dn_fq', 'l_br_to', 'l_br_who', 'l_ln_to', 'l_ln_who', 'l_dn_to',
    'l_dn_who', 'l_out_fq', 'ls_1yr', 'lk_edu', 'lk_lb_co', 'lk_lb_us', 'n_diet', 'n_diet_why',
    'n_dt_etc', 'n_wat_c', 'n_en', 'n_prot', 'n_fat', 'n_chol', 'n_cho', 'n_tdf', 'n_sugar',
    'n_na', 'lf_buyer', 'lf_safe', 'dj4_dg', 'dj4_ag', 'dj4_3', 'dl1_dg', 'dl1_ag', 'dj8_dg',
    'dj8_ag', '고혈압', '당뇨', '이상지질혈증', '신장질환'
]

len(columns_to_select)

250

In [ ]:
df_fin = df[df.isnull().sum(axis=1) <= 150]
df_fin.shape

(55989, 258)

# 컬럼명 한글로 바꾸기

In [ ]:
korean_columns = [
    "개인ID", "가구ID", "조사연도", "시도", "동읍", "아파트구분", "성별", "나이", "소득4분위", "가구소득4분위",
    "소득5분위", "가구소득5분위", "교육수준_재분류", "직업코드", "가구원수", "가구세대구성", "기초생활수급여부",
    "주택소유여부", "주택형태", "가구총소득", "월평균가구총소득", "결혼여부", "결혼상태", "건강보험종류",
    "민간의료보험가입여부", "아버지아이디", "어머니아이디", "주관적건강인지", "2주내몸이불편했는지", "불편한일수",
    "독감예방접종여부", "2년내암검진여부", "1년내손상발생여부", "사고횟수", "사고연도1", "사고월1", "사고치료방법1",
    "사고치료방법_상세1", "사고연도2", "사고월2", "사고치료방법2", "사고치료방법_상세2", "사고연도3", "사고월3",
    "사고치료방법3", "사고치료방법_상세3", "학력", "부_교육수준", "모_교육수준", "경제활동상태", "직업대분류",
    "종사상지위", "임금근로자_상세", "정규직여부", "주당평균근로시간", "최장일자리_직업분류", "최장일자리_종사상지위",
    "최장일자리_임금근로자상세", "주관적체형인식", "1년간체중변화여부", "1년간체중감소량", "1년간체중증가량",
    "1년간체중조절여부", "음주경험", "음주시작연령", "1년간음주빈도", "회당음주량_범주", "폭음빈도", "주변의절주권고여부",
    "성인_1년내음주문제상담여부", "안전벨트착용여부_운전시", "안전벨트착용여부_앞좌석", "음주운전여부_자전거",
    "음주운전횟수_자전거", "음주운전여부_오토바이", "음주운전횟수_오토바이", "음주운전여부_자동차",
    "음주운전횟수_자동차", "음주운전차량동승횟수", "스트레스인지정도", "2주이상연속우울감여부", "1년간자살생각여부",
    "1년간자살계획여부", "1년간자살시도여부", "1년간정신상담여부", "흡연경험여부", "흡연시작연령", "현재흡연여부",
    "하루평균흡연량", "라이트스모커_1달간흡연일수", "과거흡연자_흡연기간_월환산", "과거흡연자_흡연기간_월",
    "과거흡연자_흡연기간_월", "과거흡연자_일평균흡연량", "과거흡연자_금연기간_월환산", "과거흡연자_금연기간_연",
    "과거흡연자_금연기간_월환산", "액상형전자담배경험여부", "액상형전자담배현재사용여부", "평생사용담배종류_머금는담배",
    "평생사용담배종류_물담배", "평생사용담배종류_시가", "평생사용담배종류_기타", "평생사용담배종류_기타상세",
    "평생사용담배종류_흡연경험없음", "현재사용담배종류_머금는담배", "현재사용담배종류_물담배", "현재사용담배종류_시가",
    "현재사용담배종류_기타", "현재사용담배종류_기타상세", "현재사용담배종류_흡연경험없음", "1년간하루이상금연여부",
    "1개월내금연계획", "직장내간접흡연노출여부", "가정내간접흡연노출여부", "공공장소내간접흡연노출여부",
    "일반담배_한두모금피운경험", "일반담배_한두모금피운연령", "1달간흡연일수", "1달간하루흡연량", "업무_고강도신체활동여부",
    "업무_고강도신체활동일수", "업무_고강도신체활동시간", "업무_고강도신체활동분", "업무_중강도신체활동여부",
    "업무_중강도신체활동일수", "업무_중강도신체활동시간", "업무_중강도신체활동분", "일상생활_신체활동여부",
    "일상생활_신체활동일수", "일상생활_신체활동시간", "일상생활_신체활동분", "여가_고강도신체활동여부",
    "여가_고강도신체활동일수", "여가_고강도신체활동시간", "여가_고강도신체활동분", "여가_중강도신체활동여부",
    "여가_중강도신체활동일수", "여가_중강도신체활동시간", "여가_중강도신체활동분", "하루앉아서보내는시간",
    "하루앉아서보내는분", "1주간걷기일수", "걷기지속시간", "걷기지속분", "1주간근력운동일수", "월경여부", "무월경사유",
    "폐경연령", "초경연령", "임신경험여부", "임신횟수", "출산경험", "첫출산연령", "마지막출산연령", "모유수유경험",
    "모유수유자녀수", "모유수유기간_월환산", "경구피임약복용여부", "생리여부", "임신여부", "임신개월수",
    "만성질환_가족력여부", "부_고혈압여부", "모_고혈압여부", "형제자매_고혈압여부", "부_고지혈증여부", "모_고지혈증여부",
    "형제자매_고지혈증여부", "부_허혈성심장질환여부", "모_허혈성심장질환여부", "형제자매_허혈성심장질환여부",
    "부_뇌졸중여부", "모_뇌졸중여부", "형제자매_뇌졸중여부", "부_당뇨여부", "모_당뇨여부", "형제자매_당뇨여부",
    "맥박규칙성여부", "키", "몸무게", "허리둘레", "체질량지수", "비만여부", "어제하루칫솔질여부", "칫솔질시기_아침전",
    "칫솔질시기_아침후", "칫솔질시기_점심전", "칫솔질시기_점심후", "칫솔질시기_저녁전", "칫솔질시기_저녁후",
    "칫솔질시기_간식후", "칫솔질시기_자기전", "씹기문제", "말하기문제", "1년내구강검진여부", "치과이용여부", "구강xray여부",
    "잇몸병치료여부", "단순충치치료여부", "치아신경치료여부", "발치/구강내수술여부", "빠지거나부러진치아치료여부",
    "보철물제작수리여부", "기타여부", "치과진료미치료여부", "치과진료미치료이유", "치과진료미치료이유_기타상세", "본인청력",
    "직업적소음노출여부", "5분이상이명경험", "최근1년주평균아침식사빈도", "최근1년주평균점심식사빈도", "최근1년주평균저녁식사빈도",
    "아침_가족외사람동반여부", "아침_가족외사람동반대상", "점심_가족외사람동반여부", "점심_가족외사람동반대상",
    "저녁_가족외사람동반여부", "저녁_가족외사람동반대상", "평균외식빈도", "2주이상식이보충제복용여부", "영양교육및상담경험여부",
    "영양표시인지여부", "영양표시이용여부", "식사요법여부", "식사요법이유", "식사요법이유_기타", "물섭취량", "에너지섭취량",
    "단백질섭취량", "지방섭취량", "콜레스테롤섭취량", "탄수화물섭취량", "식이섬유섭취량", "당섭취량", "나트륨섭취량",
    "가구내식생활관리자여부", "식생활형편", "천식의사진단여부", "천식진단시기", "천식약복용", "아토피의사진단여부",
    "아토피진단시기", "비염의사진단여부", "비염진단시기", "고혈압", "당뇨병", "이상지질혈증", "신장질환"
]
len(korean_columns)

250

In [ ]:
# 컬럼 매핑 딕셔너리 생성
columns_name_map = dict(zip(columns_to_select, korean_columns))

# 필요한 컬럼만 선택
df = df_fin[columns_to_select]

# 컬럼명을 한글로 변경
df.rename(columns=columns_name_map, inplace=True)
df.head()

<ipython-input-7-f31b29dabd25>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.rename(columns=columns_name_map, inplace=True)


개인ID      가구ID    조사연도   시도   동읍  아파트구분   성별    나이  소득4분위  가구소득4분위  \
0  A651172801  A6511728  2016.0  1.0  1.0    2.0  1.0  61.0    3.0      3.0   
1  A651172802  A6511728  2016.0  1.0  1.0    2.0  2.0  57.0    3.0      3.0   
2  A651183001  A6511830  2016.0  1.0  1.0    2.0  2.0  41.0    3.0      3.0   
3  A651183002  A6511830  2016.0  1.0  1.0    2.0  1.0  39.0    3.0      3.0   
4  A651183003  A6511830  2016.0  1.0  1.0    2.0  2.0  38.0    3.0      3.0   

   ...  천식진단시기  천식약복용  아토피의사진단여부  아토피진단시기  비염의사진단여부  비염진단시기  고혈압  당뇨병  이상지질혈증  \
0  ...   888.0    8.0        0.0    888.0       1.0    51.0  1.0  1.0     0.0   
1  ...   888.0    8.0        0.0    888.0       0.0   888.0  0.0  0.0     0.0   
2  ...   888.0    8.0        0.0    888.0       0.0   888.0  0.0  0.0     0.0   
3  ...   888.0    8.0        0.0    888.0       1.0    11.0  1.0  0.0     1.0   
4  ...   888.0    8.0        0.0    888.0       1.0     3.0  0.0  0.0     0.0   

   신장질환  
0   0.0  
1   0.0  
2   0.0  
3   0.0  
4   0.0  

[5 rows x 250 columns]

# 전처리 시작 (정현)

## train, test 분할

In [ ]:
my_columns = [
    "시도", "동읍", "아파트구분", "성별", "나이", "소득4분위", "가구소득4분위",
    "소득5분위", "가구소득5분위", "교육수준_재분류", "가구원수", "가구세대구성", "기초생활수급여부",
    "주택소유여부", "주택형태", "가구총소득", "월평균가구총소득", "결혼여부", "결혼상태", "건강보험종류",
    "민간의료보험가입여부", "최근1년주평균아침식사빈도", "최근1년주평균점심식사빈도", "최근1년주평균저녁식사빈도",
    "아침_가족외사람동반여부", "아침_가족외사람동반대상", "점심_가족외사람동반여부", "점심_가족외사람동반대상",
    "저녁_가족외사람동반여부", "저녁_가족외사람동반대상", "평균외식빈도", "2주이상식이보충제복용여부", "영양교육및상담경험여부",
    "영양표시인지여부", "영양표시이용여부", "식사요법여부", "식사요법이유", "식사요법이유_기타", "물섭취량", "에너지섭취량",
    "단백질섭취량", "지방섭취량", "콜레스테롤섭취량", "탄수화물섭취량", "식이섬유섭취량", "당섭취량", "나트륨섭취량",
    "가구내식생활관리자여부", "식생활형편", "천식의사진단여부", "천식진단시기", "천식약복용", "아토피의사진단여부",
    "아토피진단시기", "비염의사진단여부", "비염진단시기", "고혈압", "당뇨병", "이상지질혈증", "신장질환"
]
len(my_columns)
df = df[my_columns]
df

시도   동읍  아파트구분   성별    나이  소득4분위  가구소득4분위  소득5분위  가구소득5분위  교육수준_재분류  \
0       1.0  1.0    2.0  1.0  61.0    3.0      3.0    4.0      4.0       4.0   
1       1.0  1.0    2.0  2.0  57.0    3.0      3.0    4.0      4.0       4.0   
2       1.0  1.0    2.0  2.0  41.0    3.0      3.0    3.0      4.0       4.0   
3       1.0  1.0    2.0  1.0  39.0    3.0      3.0    4.0      4.0       4.0   
4       1.0  1.0    2.0  2.0  38.0    3.0      3.0    4.0      4.0       4.0   
...     ...  ...    ...  ...   ...    ...      ...    ...      ...       ...   
55984  16.0  1.0    2.0  1.0  68.0    1.0      1.0    1.0      1.0       1.0   
55985  16.0  1.0    2.0  1.0  76.0    3.0      2.0    4.0      2.0       1.0   
55986  16.0  1.0    2.0  2.0  62.0    1.0      1.0    1.0      1.0       1.0   
55987  16.0  1.0    2.0  1.0  66.0    1.0      1.0    2.0      1.0       1.0   
55988  16.0  1.0    2.0  2.0  80.0    3.0      1.0    3.0      1.0       1.0   

       ...  천식진단시기  천식약복용  아토피의사진단여부  아토피진단시기  비염의사진단여부  비염진단시기  고혈압  당뇨병  \
0      ...   888.0    8.0        0.0    888.0       1.0    51.0  1.0  1.0   
1      ...   888.0    8.0        0.0    888.0       0.0   888.0  0.0  0.0   
2      ...   888.0    8.0        0.0    888.0       0.0   888.0  0.0  0.0   
3      ...   888.0    8.0        0.0    888.0       1.0    11.0  1.0  0.0   
4      ...   888.0    8.0        0.0    888.0       1.0     3.0  0.0  0.0   
...    ...     ...    ...        ...      ...       ...     ...  ...  ...   
55984  ...   888.0    8.0        1.0     68.0       1.0    68.0  0.0  0.0   
55985  ...   888.0    8.0        0.0    888.0       0.0   888.0  0.0  1.0   
55986  ...   888.0    8.0        0.0    888.0       0.0   888.0  1.0  1.0   
55987  ...   888.0    8.0        0.0    888.0       0.0   888.0  1.0  1.0   
55988  ...    80.0    3.0        0.0    888.0       0.0   888.0  1.0  0.0   

       이상지질혈증  신장질환  
0         0.0   0.0  
1         0.0   0.0  
2         0.0   0.0  
3         1.0   0.0  
4         0.0   0.0  
...       ...   ...  
55984     1.0   0.0  
55985     0.0   0.0  
55986     1.0   0.0  
55987     1.0   0.0  
55988     1.0   0.0  

[55989 rows x 60 columns]

In [ ]:
df = df.drop(columns=['식사요법이유_기타'])

In [ ]:
# 예시: 데이터프레임 df에서 타겟 컬럼 추출 (각 타겟 컬럼은 0과 1로 이루어짐)
target_columns = ['고혈압', '당뇨병', '이상지질혈증', '신장질환']

# X는 특징 데이터 (타겟 컬럼 제외)
x = df.drop(columns=target_columns)

# y는 타겟 변수들 (각각 0과 1로 이루어짐)
y = df[target_columns].values

# MultilabelStratifiedKFold 객체 생성 (n_splits=1로 8:2 비율로 분할)
mskf = MultilabelStratifiedKFold(n_splits=5, random_state=42, shuffle=True)

# 층화 샘플링을 통한 train, test 데이터 나누기
for train_idx, test_idx in mskf.split(x, y):
    df, test = x.iloc[train_idx], x.iloc[test_idx]
    y_train, y_test = y[train_idx], y[test_idx]
    break

# 결과 확인
print("train", df.shape)
print("test", test.shape)

train (44791, 55)
test (11198, 55)


In [ ]:
test_idx[:20]

array([ 19,  22,  24,  28,  53,  60,  63,  64,  65,  67,  72,  73,  76,
        78,  79,  83,  98, 109, 127, 131])

In [ ]:
test

시도   동읍  아파트구분   성별    나이  소득4분위  가구소득4분위  소득5분위  가구소득5분위  교육수준_재분류  \
19      1.0  1.0    2.0  2.0  36.0    4.0      4.0    5.0      5.0       4.0   
22      1.0  1.0    2.0  2.0  40.0    4.0      4.0    5.0      5.0       4.0   
24      1.0  1.0    2.0  2.0  46.0    4.0      4.0    5.0      5.0       4.0   
28      1.0  1.0    2.0  2.0  40.0    2.0      3.0    2.0      3.0       4.0   
53      1.0  1.0    2.0  2.0  63.0    1.0      1.0    1.0      1.0       3.0   
...     ...  ...    ...  ...   ...    ...      ...    ...      ...       ...   
55955  16.0  1.0    1.0  2.0  20.0    4.0      4.0    5.0      4.0       3.0   
55958  16.0  1.0    1.0  2.0  34.0    2.0      3.0    3.0      3.0       3.0   
55964  16.0  1.0    1.0  1.0  64.0    4.0      4.0    4.0      5.0       3.0   
55967  16.0  1.0    1.0  1.0  52.0    2.0      3.0    3.0      3.0       4.0   
55985  16.0  1.0    2.0  1.0  76.0    3.0      2.0    4.0      2.0       1.0   

       ...       나트륨섭취량  가구내식생활관리자여부  식생활형편  천식의사진단여부  천식진단시기  천식약복용  \
19     ...  2653.384827          1.0    1.0       0.0   888.0    8.0   
22     ...  3086.591741          1.0    1.0       0.0   888.0    8.0   
24     ...  3352.292780          1.0    1.0       0.0   888.0    8.0   
28     ...  3762.208596          1.0    2.0       0.0   888.0    8.0   
53     ...  1318.591841          1.0    1.0       0.0   888.0    8.0   
...    ...          ...          ...    ...       ...     ...    ...   
55955  ...   779.534091          1.0    2.0       0.0   888.0    8.0   
55958  ...  5191.259632          1.0    2.0       0.0   888.0    8.0   
55964  ...  2415.187373          1.0    1.0       0.0   888.0    8.0   
55967  ...  3491.780774          1.0    2.0       0.0   888.0    8.0   
55985  ...  4615.603131          2.0    2.0       0.0   888.0    8.0   

       아토피의사진단여부  아토피진단시기  비염의사진단여부  비염진단시기  
19           0.0    888.0       1.0    15.0  
22           0.0    888.0       0.0   888.0  
24           0.0    888.0       1.0    25.0  
28           0.0    888.0       0.0   888.0  
53           0.0    888.0       0.0   888.0  
...          ...      ...       ...     ...  
55955        0.0    888.0       0.0   888.0  
55958        0.0    888.0       0.0   888.0  
55964        0.0    888.0       0.0   888.0  
55967        0.0    888.0       0.0   888.0  
55985        0.0    888.0       0.0   888.0  

[11198 rows x 55 columns]

# 추가컬럼들

## 나이범주

In [ ]:
# 나이를 10년 단위로 범주화
bins = list(range(10, 90, 10)) + [float('inf')]  # 10 단위로 나누고, 100+는 별도 범주
labels = [f"{i}대" for i in range(10, 90, 10)]
df["나이범주"] = pd.cut(df["나이"], bins=bins, labels=labels, right=False)

<ipython-input-16-4de1fe182d2e>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["나이범주"] = pd.cut(df["나이"], bins=bins, labels=labels, right=False)


In [ ]:
test["나이범주"] = pd.cut(test["나이"], bins=bins, labels=labels, right=False)

<ipython-input-17-8070b3442bb3>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["나이범주"] = pd.cut(test["나이"], bins=bins, labels=labels, right=False)


## 11~14 (소득분위수)
- 4분위수의 경우
  - 1:하, 2:중하, 3:중상, 4:상
- 5분위수의 경우
  - 1:하, 2:중하, 3:중, 4:중상, 5:상

- 소득4분위수

In [ ]:
df["소득4분위"] = df["소득4분위"].fillna(9)
test["소득4분위"] = test["소득4분위"].fillna(9)

<ipython-input-18-426e691680ac>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["소득4분위"] = df["소득4분위"].fillna(9)
<ipython-input-18-426e691680ac>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["소득4분위"] = test["소득4분위"].fillna(9)


- 가구소득4분위수

In [ ]:
df["가구소득4분위"] = df["가구소득4분위"].fillna(9)
test["가구소득4분위"] = test["가구소득4분위"].fillna(9)

<ipython-input-19-d02e16352639>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["가구소득4분위"] = df["가구소득4분위"].fillna(9)
<ipython-input-19-d02e16352639>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["가구소득4분위"] = test["가구소득4분위"].fillna(9)


- 소득5분위수

In [ ]:
df["소득5분위"] = df["소득5분위"].fillna(9)
test["소득5분위"] = test["소득5분위"].fillna(9)

<ipython-input-20-7a6b7e486832>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["소득5분위"] = df["소득5분위"].fillna(9)
<ipython-input-20-7a6b7e486832>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["소득5분위"] = test["소득5분위"].fillna(9)


- 가구소득5분위수

In [ ]:
df["가구소득5분위"] = df["가구소득5분위"].fillna(9)
test["가구소득5분위"] = test["가구소득5분위"].fillna(9)

<ipython-input-21-86f8e5da6fb5>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["가구소득5분위"] = df["가구소득5분위"].fillna(9)
<ipython-input-21-86f8e5da6fb5>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["가구소득5분위"] = test["가구소득5분위"].fillna(9)


## 15 (교육수준_재분류)

In [ ]:
# 나이범주별 최빈값 계산
mode_by_age_group = df.groupby("나이범주")["교육수준_재분류"].apply(lambda x: x.mode().iloc[0] if not x.mode().empty else None)

# 결측값 채우기
df["교육수준_재분류"] = df.apply(
    lambda row: mode_by_age_group[row["나이범주"]] if pd.isnull(row["교육수준_재분류"]) else row["교육수준_재분류"], axis=1
)

<ipython-input-22-81e812c48b28>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  mode_by_age_group = df.groupby("나이범주")["교육수준_재분류"].apply(lambda x: x.mode().iloc[0] if not x.mode().empty else None)
<ipython-input-22-81e812c48b28>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["교육수준_재분류"] = df.apply(


In [ ]:
test["교육수준_재분류"] = test.apply(
    lambda row: mode_by_age_group[row["나이범주"]] if pd.isnull(row["교육수준_재분류"]) else row["교육수준_재분류"], axis=1
)

<ipython-input-23-04a4f1365b07>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["교육수준_재분류"] = test.apply(


In [ ]:
df["교육수준_재분류"].isnull().sum(), test["교육수준_재분류"].isnull().sum()

(0, 0)

## 19 (기초생활수급 여부)
- 10: 수급경험 있음(과거 또는 현재), 20: 수급경험 없음, 99: 모름/무응답
- 모름/무응답 중 소득4분위수가 하로 나오는 사람들 기초생활수급여부에 예로 바꿈. 나머지는 그대로 무응답처리

In [ ]:
df["기초생활수급여부"].value_counts()

기초생활수급여부
20.0    41747
10.0     3014
99.0       30
Name: count, dtype: int64

In [ ]:
df["기초생활수급여부"].isnull().sum()

0

In [ ]:
# "기초생활수급여부" 열이 99인 행에서 "소득4분위수" 값 확인
welfare_99_income_quartile = df.loc[df['기초생활수급여부'] == 99, '소득4분위']

# 결과 출력
welfare_99_income_quartile.value_counts()

소득4분위
1.0    17
9.0     6
2.0     4
3.0     2
4.0     1
Name: count, dtype: int64

In [ ]:
condition = (df['기초생활수급여부'] == 99) & (df['소득4분위'] == 1)
df.loc[condition, '기초생활수급여부'] = 10


In [ ]:
condition = (test['기초생활수급여부'] == 99) & (test['소득4분위'] == 1)
test.loc[condition, '기초생활수급여부'] = 10

## 24 (월평균가구총소득)
- 결측치 모름/무응답 처리
- 가구총소득에서 모름/무응답으로 답한 사람들이 결측치로 나타나는듯

In [ ]:
# "월평균가구총소득" 열의 결측치를 999999로 처리
df['월평균가구총소득'] = df['월평균가구총소득'].fillna(999999)
df['월평균가구총소득'].isnull().sum()

<ipython-input-30-109b4db1a9f4>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['월평균가구총소득'] = df['월평균가구총소득'].fillna(999999)


0

In [ ]:
test['월평균가구총소득'] = test['월평균가구총소득'].fillna(999999)
test['월평균가구총소득'].isnull().sum()

<ipython-input-31-3374a1f16ee1>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test['월평균가구총소득'] = test['월평균가구총소득'].fillna(999999)


0

## 243~245
- 거의 안한다(주0회): 1, 주 1-2회: 2, 주 3-4회: 3, 주 5-7회: 4 로 순서형 유지하면서 매핑 다시 시켜야함.
- 모름/무응답 항목이 결측치로 빠져서 항목을 새로 만들어서 모름/무응답 처리해야함. (모름/무응답:9)

In [ ]:
reorder_mapping = {
    1: 4,  # 주 5-7회 -> 4
    2: 3,  # 주 3-4회 -> 3
    3: 2,  # 주 1-2회 -> 2
    4: 1   # 거의 안한다(주0회) -> 1
}

# 결측치 처리
df["최근1년주평균아침식사빈도"] = df["최근1년주평균아침식사빈도"].map(reorder_mapping)
df["최근1년주평균아침식사빈도"] = df["최근1년주평균아침식사빈도"].fillna(9)

<ipython-input-32-82f220087f58>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균아침식사빈도"] = df["최근1년주평균아침식사빈도"].map(reorder_mapping)
<ipython-input-32-82f220087f58>:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균아침식사빈도"] = df["최근1년주평균아침식사빈도"].fillna(9)


In [ ]:
# 결측치 처리
test["최근1년주평균아침식사빈도"] = test["최근1년주평균아침식사빈도"].map(reorder_mapping)
test["최근1년주평균아침식사빈도"] = test["최근1년주평균아침식사빈도"].fillna(9)

<ipython-input-33-91b733151942>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균아침식사빈도"] = test["최근1년주평균아침식사빈도"].map(reorder_mapping)
<ipython-input-33-91b733151942>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균아침식사빈도"] = test["최근1년주평균아침식사빈도"].fillna(9)


In [ ]:
df["최근1년주평균아침식사빈도"].isnull().sum(), test["최근1년주평균아침식사빈도"].isnull().sum()

(0, 0)

In [ ]:
reorder_mapping = {
    1: 4,  # 주 5-7회 -> 4
    2: 3,  # 주 3-4회 -> 3
    3: 2,  # 주 1-2회 -> 2
    4: 1   # 거의 안한다(주0회) -> 1
}

# 결측치 처리
df["최근1년주평균점심식사빈도"] = df["최근1년주평균점심식사빈도"].map(reorder_mapping)
df["최근1년주평균점심식사빈도"] = df["최근1년주평균점심식사빈도"].fillna(9)

<ipython-input-35-994803196ed1>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균점심식사빈도"] = df["최근1년주평균점심식사빈도"].map(reorder_mapping)
<ipython-input-35-994803196ed1>:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균점심식사빈도"] = df["최근1년주평균점심식사빈도"].fillna(9)


In [ ]:
# 결측치 처리
test["최근1년주평균점심식사빈도"] = test["최근1년주평균점심식사빈도"].map(reorder_mapping)
test["최근1년주평균점심식사빈도"] = test["최근1년주평균점심식사빈도"].fillna(9)

<ipython-input-36-ef87eba35bf9>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균점심식사빈도"] = test["최근1년주평균점심식사빈도"].map(reorder_mapping)
<ipython-input-36-ef87eba35bf9>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균점심식사빈도"] = test["최근1년주평균점심식사빈도"].fillna(9)


In [ ]:
df["최근1년주평균점심식사빈도"].isnull().sum(), test["최근1년주평균점심식사빈도"].isnull().sum()

(0, 0)

In [ ]:
reorder_mapping = {
    1: 4,  # 주 5-7회 -> 4
    2: 3,  # 주 3-4회 -> 3
    3: 2,  # 주 1-2회 -> 2
    4: 1   # 거의 안한다(주0회) -> 1
}

# 결측치 처리
df["최근1년주평균저녁식사빈도"] = df["최근1년주평균저녁식사빈도"].map(reorder_mapping)
df["최근1년주평균저녁식사빈도"] = df["최근1년주평균저녁식사빈도"].fillna(9)

<ipython-input-38-f650856e3dbc>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균저녁식사빈도"] = df["최근1년주평균저녁식사빈도"].map(reorder_mapping)
<ipython-input-38-f650856e3dbc>:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["최근1년주평균저녁식사빈도"] = df["최근1년주평균저녁식사빈도"].fillna(9)


In [ ]:
# 결측치 처리
test["최근1년주평균저녁식사빈도"] = test["최근1년주평균저녁식사빈도"].map(reorder_mapping)
test["최근1년주평균저녁식사빈도"] = test["최근1년주평균저녁식사빈도"].fillna(9)

<ipython-input-39-4410ccbdd94c>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균저녁식사빈도"] = test["최근1년주평균저녁식사빈도"].map(reorder_mapping)
<ipython-input-39-4410ccbdd94c>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["최근1년주평균저녁식사빈도"] = test["최근1년주평균저녁식사빈도"].fillna(9)


## 246, 248, 250
- 다중분류(1: 예. 2: 아니오, 3: 비해당 (아침식사빈도가 주2회이하), 9: 모름/무응답) 로 항목이 이루어져있음.
- 비해당 항목이 "아침식사빈도 주2회 이하랑" 일치하는지 확인
- 결측치는 모름/무응답 처리

- 246

In [ ]:
# 결측치 처리
df["아침_가족외사람동반여부"] = df["아침_가족외사람동반여부"].fillna(9)
test["아침_가족외사람동반여부"] = test["아침_가족외사람동반여부"].fillna(9)

<ipython-input-40-b774db8d3a16>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["아침_가족외사람동반여부"] = df["아침_가족외사람동반여부"].fillna(9)
<ipython-input-40-b774db8d3a16>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["아침_가족외사람동반여부"] = test["아침_가족외사람동반여부"].fillna(9)


- 248

In [ ]:
# 결측치 처리
df["점심_가족외사람동반여부"] = df["점심_가족외사람동반여부"].fillna(9)
test["점심_가족외사람동반여부"] = test["점심_가족외사람동반여부"].fillna(9)

<ipython-input-41-4489e5948a6e>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["점심_가족외사람동반여부"] = df["점심_가족외사람동반여부"].fillna(9)
<ipython-input-41-4489e5948a6e>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["점심_가족외사람동반여부"] = test["점심_가족외사람동반여부"].fillna(9)


- 250

In [ ]:
# 결측치 처리
df["저녁_가족외사람동반여부"] = df["저녁_가족외사람동반여부"].fillna(9)
test["저녁_가족외사람동반여부"] = test["저녁_가족외사람동반여부"].fillna(9)

<ipython-input-42-55d87efec6e2>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["저녁_가족외사람동반여부"] = df["저녁_가족외사람동반여부"].fillna(9)
<ipython-input-42-55d87efec6e2>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["저녁_가족외사람동반여부"] = test["저녁_가족외사람동반여부"].fillna(9)


## 247, 249, 251
- 다중분류(1: 예. 2: 아니오, 8: 비해당 (동반여부가 "아니오"/"비해당" 인 경우), 9: 모름/무응답) 로 항목이 이루어져있음.
- 결측치 무응답으로

- 247

In [ ]:
# 결측치 처리
df["아침_가족외사람동반대상"] = df["아침_가족외사람동반대상"].fillna(9)
test["아침_가족외사람동반대상"] = test["아침_가족외사람동반대상"].fillna(9)

<ipython-input-43-8b7141258373>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["아침_가족외사람동반대상"] = df["아침_가족외사람동반대상"].fillna(9)
<ipython-input-43-8b7141258373>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["아침_가족외사람동반대상"] = test["아침_가족외사람동반대상"].fillna(9)


- 249

In [ ]:
df["점심_가족외사람동반대상"] = df["점심_가족외사람동반대상"].fillna(9)
test["점심_가족외사람동반대상"] = test["점심_가족외사람동반대상"].fillna(9)

<ipython-input-44-4c46c12ef522>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["점심_가족외사람동반대상"] = df["점심_가족외사람동반대상"].fillna(9)
<ipython-input-44-4c46c12ef522>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["점심_가족외사람동반대상"] = test["점심_가족외사람동반대상"].fillna(9)


- 251

In [ ]:
df["저녁_가족외사람동반대상"] = df["저녁_가족외사람동반대상"].fillna(9)
test["저녁_가족외사람동반대상"] = test["저녁_가족외사람동반대상"].fillna(9)

<ipython-input-45-f503704d191a>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["저녁_가족외사람동반대상"] = df["저녁_가족외사람동반대상"].fillna(9)
<ipython-input-45-f503704d191a>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["저녁_가족외사람동반대상"] = test["저녁_가족외사람동반대상"].fillna(9)


## 252 (평균외식빈도)
- 1: 하루 2회 이상, 2: 하루 1회, 3: 주 5-6회, 4: 주 3-4회, 5: 주 1-2회, 6: 월 1-3회, 7: 거의 안 한다 (월1회 미만), 9: 모름/무응답 으로 이루어져있음.
- 순서형 유지하면서 외식빈도 가끔 -> 외식빈도 잦음 순으로 다시 매핑 적용시키기
- 결측치 무응답 처리

In [ ]:
mapping = {
    1: 7,  # 하루 2회 이상 -> 7
    2: 6,  # 하루 1회 -> 6
    3: 5,  # 주 5-6회 -> 5
    4: 4,  # 주 3-4회 -> 4
    5: 3,  # 주 1-2회 -> 3
    6: 2,  # 월 1-3회 -> 2
    7: 1   # 거의 안 한다 (월 1회 미만) -> 1
}

# 역순 매핑 적용
df["평균외식빈도"] = df["평균외식빈도"].map(mapping)

<ipython-input-46-7a99c2df2053>:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["평균외식빈도"] = df["평균외식빈도"].map(mapping)


In [ ]:
# 역순 매핑 적용
test["평균외식빈도"] = test["평균외식빈도"].map(mapping)

<ipython-input-47-32af7ca013a8>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["평균외식빈도"] = test["평균외식빈도"].map(mapping)


In [ ]:
# 결측치 처리
df["평균외식빈도"] = df["평균외식빈도"].fillna(9)
test["평균외식빈도"] = test["평균외식빈도"].fillna(9)

<ipython-input-48-fa1495a0848a>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["평균외식빈도"] = df["평균외식빈도"].fillna(9)
<ipython-input-48-fa1495a0848a>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["평균외식빈도"] = test["평균외식빈도"].fillna(9)


## 253 (2주 이상 식이보충제 복용여부)
- 다중분류(1: 예, 2:아니오, 9:모름/무응답) 피설문자들의 나이가 대체로 노령임.
- 결측치를 그룹별(나이 or 성별) 최빈값으로 채우는 방식으로 진행
- 예:1, 아니오:0 로 변환

In [ ]:
# 나이범주별 2주 이상 식이보충제 복용여부 그룹화 및 개수 확인
result = df.groupby(["나이범주", "2주이상식이보충제복용여부"], observed=False).size().reset_index(name="count")

In [ ]:
df["나이범주"].isnull().sum()

0

In [ ]:
result

나이범주  2주이상식이보충제복용여부  count
0   10대            1.0     91
1   10대            2.0    275
2   20대            1.0   1775
3   20대            2.0   2311
4   30대            1.0   3186
5   30대            2.0   2420
6   40대            1.0   4194
7   40대            2.0   2641
8   50대            1.0   4636
9   50대            2.0   2720
10  60대            1.0   4835
11  60대            2.0   2828
12  70대            1.0   3119
13  70대            2.0   2498
14  80대            1.0    919
15  80대            2.0    917

In [ ]:
import random
# 조건에 맞게 결측치 처리
random.seed(42)
def fill_missing(row):
    if pd.isna(row["2주이상식이보충제복용여부"]):
        if row["나이"] >= 80:
            # 80대 이상인 경우, 50% 확률로 1: 예 또는 2: 아니오 할당
            return random.choices([1, 2], weights=[50, 50])[0]
        elif row["나이"] >= 30:
            # 30대 이상인 경우, 1: 예 65%, 2: 아니오 35% 확률로 할당
            return random.choices([1, 2], weights=[65, 35])[0]
        else:
            # 30대 미만인 경우, 1: 예 40%, 2: 아니오 60% 확률로 할당
            return random.choices([1, 2], weights=[40, 60])[0]
    return row["2주이상식이보충제복용여부"]

random.seed(42)
df["2주이상식이보충제복용여부"] = df.apply(fill_missing, axis=1)
random.seed(42)
test["2주이상식이보충제복용여부"] = test.apply(fill_missing, axis=1)

<ipython-input-52-9d1973f65b5f>:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["2주이상식이보충제복용여부"] = df.apply(fill_missing, axis=1)
<ipython-input-52-9d1973f65b5f>:20: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["2주이상식이보충제복용여부"] = test.apply(fill_missing, axis=1)


In [ ]:
df["2주이상식이보충제복용여부"] = df["2주이상식이보충제복용여부"].replace({
    1: 1,
    2: 0
    })

<ipython-input-53-b860d782fd9e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["2주이상식이보충제복용여부"] = df["2주이상식이보충제복용여부"].replace({


In [ ]:
test["2주이상식이보충제복용여부"] = test["2주이상식이보충제복용여부"].replace({
    1: 1,
    2: 0
    })

<ipython-input-54-db06e562bdaf>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["2주이상식이보충제복용여부"] = test["2주이상식이보충제복용여부"].replace({


## 254, 255
- 다중분류(1: 예, 2: 아니오, 3: 비해당(초등학생 미만), 9: 모름/무응답)으로 이루어짐
- 결측치 2: 아니오로 처리하는게 나을듯

- 254

In [ ]:
# 결측치를 2: 아니오로 처리
df["영양교육및상담경험여부"] = df["영양교육및상담경험여부"].fillna(2)
test["영양교육및상담경험여부"] = test["영양교육및상담경험여부"].fillna(2)

<ipython-input-55-bd7786828cfb>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["영양교육및상담경험여부"] = df["영양교육및상담경험여부"].fillna(2)
<ipython-input-55-bd7786828cfb>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["영양교육및상담경험여부"] = test["영양교육및상담경험여부"].fillna(2)


- 255

In [ ]:
# "나이범주"와 "영양표시 인지여부"를 기준으로 그룹화하여 빈도 수 계산
result = df.groupby(["나이범주", "영양표시인지여부"]).size().reset_index(name="count")
result

<ipython-input-56-86e19f052731>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  result = df.groupby(["나이범주", "영양표시인지여부"]).size().reset_index(name="count")


나이범주  영양표시인지여부  count
0   10대       1.0    340
1   10대       2.0     25
2   10대       9.0      1
3   20대       1.0   3878
4   20대       2.0    208
5   20대       9.0      0
6   30대       1.0   5268
7   30대       2.0    338
8   30대       9.0      0
9   40대       1.0   6278
10  40대       2.0    555
11  40대       9.0      2
12  50대       1.0   6018
13  50대       2.0   1336
14  50대       9.0      2
15  60대       1.0   4941
16  60대       2.0   2716
17  60대       9.0      6
18  70대       1.0   2113
19  70대       2.0   3503
20  70대       9.0      1
21  80대       1.0    386
22  80대       2.0   1450
23  80대       9.0      0

In [ ]:
random.seed(42)
def fill_missing(row):
    if pd.isna(row["영양표시인지여부"]):
        if row["나이범주"] in ["20대", "30대", "40대"]:
            # 50대 미만인 경우, 1: 예 90%, 2: 아니오 10%
            return random.choices([1, 2], weights=[90, 10])[0]
        elif row["나이범주"] == "50대":
            # 50대인 경우, 1: 예 85%, 2: 아니오 15%
            return random.choices([1, 2], weights=[85, 15])[0]
        elif row["나이범주"] == "60대":
            # 60대인 경우, 1: 예 65%, 2: 아니오 35%
            return random.choices([1, 2], weights=[65, 35])[0]
        elif row["나이범주"] in ["70대", "80대"]:
            # 70대 이상인 경우, 1: 예 20%, 2: 아니오 80%
            return random.choices([1, 2], weights=[20, 80])[0]
    return row["영양표시인지여부"]
random.seed(42)
df["영양표시인지여부"] = df.apply(fill_missing, axis=1)

random.seed(42)
test["영양표시인지여부"] = test.apply(fill_missing, axis=1)

<ipython-input-57-29976ab565bf>:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["영양표시인지여부"] = df.apply(fill_missing, axis=1)
<ipython-input-57-29976ab565bf>:21: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["영양표시인지여부"] = test.apply(fill_missing, axis=1)


In [ ]:
# 나머지 결측치 9: 모름/무응답 처리
df["영양표시인지여부"] = df["영양표시인지여부"].fillna(9)
test["영양표시인지여부"] = test["영양표시인지여부"].fillna(9)

<ipython-input-58-69dbcc14ee8a>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["영양표시인지여부"] = df["영양표시인지여부"].fillna(9)
<ipython-input-58-69dbcc14ee8a>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["영양표시인지여부"] = test["영양표시인지여부"].fillna(9)


## 256 (영양표시이용여부)
- 다중분류(1: 예, 2: 아니오, 8: 비해당(초등학생 미만), 9: 모름/무응답)
- 결측치 아니오로 대체해도 될듯

In [ ]:
# "영양표시이용여부"가 결측치가 아닌 사람들에 대해 "나이범주"와 "영양표시이용여부"의 합산
grouped = df[df["영양표시이용여부"].notna()].groupby(["나이범주", "영양표시이용여부"]).size().reset_index(name="count")

# 결과 확인
grouped

<ipython-input-59-d6a6ce24410c>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  grouped = df[df["영양표시이용여부"].notna()].groupby(["나이범주", "영양표시이용여부"]).size().reset_index(name="count")


나이범주  영양표시이용여부  count
0   10대       1.0    116
1   10대       2.0    224
2   10대       8.0     25
3   10대       9.0      1
4   20대       1.0   1739
5   20대       2.0   2139
6   20대       8.0    208
7   20대       9.0      0
8   30대       1.0   2407
9   30대       2.0   2861
10  30대       8.0    338
11  30대       9.0      0
12  40대       1.0   2687
13  40대       2.0   3591
14  40대       8.0    555
15  40대       9.0      2
16  50대       1.0   2003
17  50대       2.0   4015
18  50대       8.0   1336
19  50대       9.0      2
20  60대       1.0   1339
21  60대       2.0   3602
22  60대       8.0   2716
23  60대       9.0      6
24  70대       1.0    413
25  70대       2.0   1700
26  70대       8.0   3503
27  70대       9.0      1
28  80대       1.0     68
29  80대       2.0    318
30  80대       8.0   1450
31  80대       9.0      0

In [ ]:
random.seed(42)
def fill_missing(row):
    if pd.isna(row["영양표시이용여부"]) or row["영양표시이용여부"] == 9:
        if row["나이범주"] == "10대":
            # 10대의 경우, 1: 예 25%, 2: 아니오 75%
            return random.choices([1, 2], weights=[25, 75])[0]
        elif row["나이범주"] in ["20대", "30대", "40대"]:
            # 50대인 경우, 1: 예 40%, 2: 아니오 60%
            return random.choices([1, 2], weights=[40, 60])[0]
        elif row["나이범주"]in ["50대", "60대"]:
            # 60대인 경우, 1: 예 65%, 2: 아니오 35%
            return random.choices([1, 2], weights=[30, 70])[0]
        elif row["나이범주"] in ["70대", "80대"]:
            # 70대 이상인 경우, 1: 예 20%, 2: 아니오 80%
            return random.choices([1, 2], weights=[20, 80])[0]
    return row["영양표시이용여부"]

random.seed(42)
df["영양표시이용여부"] = df.apply(fill_missing, axis=1)

random.seed(42)
test["영양표시이용여부"] = test.apply(fill_missing, axis=1)

<ipython-input-60-73456d27d8cd>:19: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["영양표시이용여부"] = df.apply(fill_missing, axis=1)
<ipython-input-60-73456d27d8cd>:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["영양표시이용여부"] = test.apply(fill_missing, axis=1)


## 259 (식사요법여부)
- 다중분류(1: 예, 2:아니오, 9:모름/무응답)
- 결측치 무응답처리

In [ ]:
grouped = df[df["식사요법여부"].notna()].groupby(["성별", "식사요법여부"]).size().reset_index(name="count")
grouped

성별  식사요법여부  count
0  1.0     1.0   3857
1  1.0     2.0  12846
2  1.0     9.0      9
3  2.0     1.0   6847
4  2.0     2.0  15756
5  2.0     9.0     21

In [ ]:
random.seed(42)
def fill_missing(row):
    if pd.isna(row["식사요법여부"]) or row["식사요법여부"] == 9:
        if row["성별"] == 1:
            # 남자인 경우, 1: 예 20%, 2: 아니오 80%
            return random.choices([1, 2], weights=[20, 80])[0]
        elif row["성별"] == 2:
            # 여자인 경우, 1: 예 30%, 2: 아니오 70%
            return random.choices([1, 2], weights=[30, 70])[0]
    return row["식사요법여부"]
random.seed(42)
df["식사요법여부"] = df.apply(fill_missing, axis=1)

random.seed(42)
test["식사요법여부"] = test.apply(fill_missing, axis=1)

<ipython-input-62-73bfc0177eb9>:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["식사요법여부"] = df.apply(fill_missing, axis=1)
<ipython-input-62-73bfc0177eb9>:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["식사요법여부"] = test.apply(fill_missing, axis=1)


In [ ]:
df["식사요법여부"].value_counts()

식사요법여부
2.0    32729
1.0    12062
Name: count, dtype: int64

In [ ]:
df["식사요법여부"].isnull().sum()

0

## 260 (식사요법이유)
- 다중분류( 1: 질환이있어서, 2: 체중을 조절하기 위해서, 3: 기타, 8: 비해당(식사요법여부:아니오), 9: 모름/무응답)

In [ ]:
# 결측치 무응답 처리
df["식사요법이유"] = df["식사요법이유"].fillna(9)
test["식사요법이유"] = test["식사요법이유"].fillna(9)

<ipython-input-65-fe568a387478>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["식사요법이유"] = df["식사요법이유"].fillna(9)
<ipython-input-65-fe568a387478>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  test["식사요법이유"] = test["식사요법이유"].fillna(9)


## 264~272
- 수치형이라 다 평균으로 대체해야지
- 성별과 나이범주별 평균으로 대체하자

In [ ]:
# "나이범주"와 "성별"별 평균 계산
grouped_mean = df.groupby(['나이범주', '성별'])[
    ["물섭취량", "에너지섭취량", "단백질섭취량", "지방섭취량", "콜레스테롤섭취량",
     "탄수화물섭취량", "식이섬유섭취량", "당섭취량", "나트륨섭취량"]].mean()
grouped_mean

<ipython-input-66-ba5c02c1890e>:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  grouped_mean = df.groupby(['나이범주', '성별'])[


물섭취량       에너지섭취량      단백질섭취량      지방섭취량    콜레스테롤섭취량  \
나이범주 성별                                                              
10대  1.0  5.665663  2630.156850  101.215080  78.576821  374.150498   
     2.0  3.869000  1685.375782   61.497465  48.308144  239.436186   
20대  1.0  6.269715  2399.603779   93.596879  70.246822  350.233912   
     2.0  4.701367  1772.186976   65.852177  53.141025  270.062638   
30대  1.0  6.499575  2528.036951   96.998186  69.234565  363.912862   
     2.0  4.656862  1751.006419   65.842570  48.252877  261.416002   
40대  1.0  5.849309  2411.028958   87.478383  58.941065  325.444650   
     2.0  4.929534  1700.153882   63.330334  43.924186  241.630345   
50대  1.0  5.666788  2297.144124   82.138507  50.278429  277.897051   
     2.0  5.149111  1653.708348   59.725024  37.530802  206.489010   
60대  1.0  5.797241  2129.703371   75.305084  41.909715  226.379117   
     2.0  4.950150  1577.543416   56.045779  31.805631  176.851644   
70대  1.0  4.869511  1858.253907   64.157437  33.012764  176.662768   
     2.0  4.436588  1404.122914   46.794336  23.957737  124.447196   
80대  1.0  4.179487  1627.571290   55.033238  27.902382  147.567233   
     2.0  3.802291  1236.948062   40.211203  19.468706   92.496612   

             탄수화물섭취량    식이섬유섭취량       당섭취량       나트륨섭취량  
나이범주 성별                                                  
10대  1.0  330.940246  21.786071  78.509089  4249.137762  
     2.0  233.575996  15.426061  58.719425  2513.921566  
20대  1.0  309.904978  22.715477  67.364578  4019.316851  
     2.0  239.444793  17.546206  59.922968  2879.068021  
30대  1.0  327.790181  25.365857  68.576589  4490.407422  
     2.0  247.329566  20.116989  56.037931  3105.122698  
40대  1.0  327.206871  27.402604  64.480278  4428.294765  
     2.0  251.630871  23.147002  58.351209  3068.490595  
50대  1.0  329.422793  30.012098  62.761108  4218.031577  
     2.0  264.287274  26.945846  62.330317  2908.598432  
60대  1.0  329.568080  31.863414  61.400366  3893.638997  
     2.0  264.432024  27.320840  55.653951  2707.587584  
70대  1.0  305.876719  28.545397  51.436360  3404.862991  
     2.0  247.912740  23.452091  45.373396  2456.218822  
80대  1.0  277.954283  24.660689  44.911382  2931.284239  
     2.0  222.445636  19.349979  36.449257  2135.685012

# 체크필요

In [ ]:
cols = ["물섭취량", "에너지섭취량", "단백질섭취량", "지방섭취량", "콜레스테롤섭취량",
     "탄수화물섭취량", "식이섬유섭취량", "당섭취량", "나트륨섭취량"]
grouped_mean = df.groupby(['나이범주', '성별'], observed=False)[cols].mean().reset_index()
df = df.merge(grouped_mean, on=['나이범주', '성별'], how='left', suffixes=('', '_median'))

# 결측치 처리
for col in cols:
    df[col] = df[col].fillna(df[f'{col}_median'])

In [ ]:
test = test.merge(grouped_mean, on=['나이범주', '성별'], how='left', suffixes=('', '_median'))

# 결측치 처리
for col in cols:
    test[col] = test[col].fillna(test[f'{col}_median'])

In [ ]:
cols_todrop = [col + "_median" for col in cols]

In [ ]:
df = df.drop(columns=cols_todrop)
test = test.drop(columns=cols_todrop)

In [ ]:
df["나트륨섭취량"].isnull().sum()

0

In [ ]:
test["물섭취량"].isnull().sum()

0

## 273 (가구 내 식생활관리자여부)
- 다중분류 (1: 예, 2: 아니오, 9: 모름/무응답)
- 결측치 무응답 처리

In [ ]:
# 결측치 무응답 처리
df["가구내식생활관리자여부"] = df["가구내식생활관리자여부"].fillna(9)
test["가구내식생활관리자여부"] = test["가구내식생활관리자여부"].fillna(9)

## 274 (식생활형편)
- 다중분류 (1: 충분한 양과 다양한 음식을 먹을 수 있었다, 2: 충분한 양의 음식으 먹을 수 있었으나, 다양한 음식은 먹지 못했다, 3: 경제적으로 어려워 가끔 먹을것이 부족했다, 4: 경제적으로 어려워 자주 먹을것이 부족했다, 9:모름/무응답)
- 결측치 무응답처리

# 체크필요

In [ ]:
tmp = df.groupby('가구소득4분위')['식생활형편'].agg(lambda x: x.mode()[0]).reset_index()
df = df.merge(tmp, on=['가구소득4분위'], how='left', suffixes=('', '_median'))
df['식생활형편'] = df['식생활형편'].fillna(df['식생활형편_median'])
df.drop(columns=['식생활형편_median'], inplace=True)

In [ ]:
test = test.merge(tmp, on=['가구소득4분위'], how='left', suffixes=('', '_median'))
test['식생활형편'] = test['식생활형편'].fillna(test['식생활형편_median'])
test.drop(columns=['식생활형편_median'], inplace=True)

In [ ]:
df.isna().sum().sum(), test.isna().sum().sum()

(0, 0)

## 나머지, target 컬럼들

- 천식의사진단여부: 0: 없음, 1: 있음, 8: 비해당(소아, 청소년), 9: 모름/무응답

In [ ]:
df["천식의사진단여부"].value_counts()

천식의사진단여부
0.0    40837
9.0     2577
1.0     1377
Name: count, dtype: int64

In [ ]:
df["천식의사진단여부"] = df["천식의사진단여부"].replace({9:0, 8:0})
test["천식의사진단여부"] = test["천식의사진단여부"].replace({9:0, 8:0})

- 천식진단시기: 0~79: 0-79세, 80: 80세 이상,
888: 비해당(소아, 청소년 의사진단 받지않음), 999: 모름/무응답

In [ ]:
df["천식진단시기"].value_counts()

천식진단시기
888.0    40837
999.0     2591
63.0        51
53.0        45
45.0        39
         ...  
74.0         7
76.0         7
39.0         6
79.0         4
26.0         4
Name: count, Length: 83, dtype: int64

In [ ]:
df["천식의사진단여부"].isnull().sum()

0

- 천식약복용
  - 1: 악화방지 및 천식증상관리를 위해 정기적으로 치료를 받음.
  - 2: 증상이 있을 때만 치료를 받음
  - 3: 치료받지 않음
  - 8: 비해당 (소아, 청소년 의사진단 받지않음)
  - 9: 모름/무응답

In [ ]:
df["천식약복용"].value_counts()

천식약복용
8.0    40837
9.0     2577
3.0      681
2.0      367
1.0      329
Name: count, dtype: int64

In [ ]:
df["천식약복용"] = df["천식약복용"].replace({
    3:0, 8:0, 9:0,
    2:1, 2:2
})
test["천식약복용"] = test["천식약복용"].replace({
    3:0, 8:0, 9:0,
    2:1, 2:2
})

- 아토피의사진단여부
  - 0: 없음
  - 1: 있음
  - 8: 비해당(소아,청소년)
  - 9: 모름/무응답

In [ ]:
df["아토피의사진단여부"].value_counts()

아토피의사진단여부
0.0    40845
9.0     2582
1.0     1364
Name: count, dtype: int64

In [ ]:
df["아토피의사진단여부"] = df["아토피의사진단여부"].replace({9:0, 8:0})
test["아토피의사진단여부"] = test["아토피의사진단여부"].replace({9:0, 8:0})

- 아토피진단시기: 0~79: 0-79세, 80: 80세 이상,
888: 비해당(소아, 청소년 의사진단 받지않음), 999: 모름/무응답

In [ ]:
df["아토피진단시기"].value_counts()

아토피진단시기
888.0    40845
999.0     2592
7.0         94
5.0         93
6.0         88
         ...  
75.0         2
74.0         2
78.0         2
76.0         1
69.0         1
Name: count, Length: 81, dtype: int64

In [ ]:
df["아토피진단시기"].isnull().sum()

0

- 비염의사진단여부
  - 0: 없음
  - 1: 있음
  - 8: 비해당(소아,청소년)
  - 9: 모름/무응답

In [ ]:
df["비염의사진단여부"].value_counts()

비염의사진단여부
0.0    35872
1.0     6336
9.0     2583
Name: count, dtype: int64

In [ ]:
df["비염의사진단여부"] = df["비염의사진단여부"].replace({9:0, 8:0})
test["비염의사진단여부"] = test["비염의사진단여부"].replace({9:0, 8:0})

- 비염진단시기: 0~79: 0-79세, 80: 80세 이상,
888: 비해당(소아, 청소년 의사진단 받지않음), 999: 모름/무응답

In [ ]:
df["비염진단시기"].value_counts()

비염진단시기
888.0    35872
999.0     2604
23.0       231
33.0       217
35.0       215
         ...  
80.0         9
0.0          8
2.0          8
1.0          5
79.0         1
Name: count, Length: 83, dtype: int64

In [ ]:
df["비염진단시기"].isnull().sum()

0

In [ ]:
df.shape, test.shape

((44791, 56), (11198, 56))

In [ ]:
df.isna().sum().sum(), test.isna().sum().sum()

(0, 0)

In [ ]:
df.shape, test.shape

((44791, 56), (11198, 56))

In [ ]:
DATA_PATH = '/content/drive/MyDrive/dayoon/데이터톤/공유 데이터셋 및 베이스라인코드/데이터 최신버젼/'

In [ ]:
df.to_csv(f'{DATA_PATH}train_kjh.csv', index=False)

In [ ]:
test.to_csv(f'{DATA_PATH}test_kjh.csv', index=False)